<launch>
  <include file="$(find pr2_overhead_grasping)/launch/init_params.launch"/>
  <include file="$(find pr2_overhead_grasping)/launch/simple_grasp_pipeline.launch"/>
  <node pkg="nodelet" type="nodelet" name="coll_manager"  args="manager" output="screen">
    <param name="num_worker_threads" type="int" value="5" />
  </node>

  <node pkg="nodelet" type="nodelet" name="sensor_filter_r" args="load collision_detection/sensor_filter coll_manager" output="screen">
    <param name="arm" type="str" value="r"/>
  </node>                 
  <node pkg="nodelet" type="nodelet" name="collision_monitor_r" args="load collision_detection/collision_monitor coll_manager" output="screen">
    <param name="arm" type="str" value="r"/>
  </node>                 
  <node pkg="nodelet" type="nodelet" name="random_forest_r" args="load collision_detection/random_forest coll_manager" output="screen">
    <param name="training_mode" type="bool" value="false"/>
    <param name="classify_topic" type="str" value="r_arm_features"/>
    <param name="results_topic" type="str" value="r_arm_collision_detect"/>
    <param name="loaded_topic" type="str" value="r_arm_cd_loaded"/>
    <param name="forest_bag_name" type="str" value="wrist_forest.bag"/>
    <param name="bag_path" value="$(find pr2_overhead_grasping)/bags/"/>
  </node>                 
  <node pkg="nodelet" type="nodelet" name="force_torque_monitor" args="load collision_detection/force_torque_monitor standalone_nodelet" output="screen">
    <param name="arm" type="str" value="r"/>
    <param name="z_thresh" type="double" value="4.0" />
    <param name="delay_time" type="double" value="0.15" />
    <param name="label" type="int" value="4" />
  </node>                 
</launch>
